app/main.py

```diff
@@ -1,34 +1,56 @@
+from typing import Optional
+
+
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(self,
+                 start: tuple,
+                 end: tuple,
+                 is_drowned: bool = False) -> None:
+        self.is_drowned = is_drowned
+        self.decks = []
+        for coord_row in range(start[0], end[0] + 1):
+            if start[0] == end[0]:
+                columns = range(start[1], end[1] + 1)
+            else:
+                columns = [start[1]]
+            for column in columns:
+                self.decks.append(Deck(coord_row, column))
```

Comment:
```suggestion
        for coord_row in range(start[0], end[0] + 1):
            for column in range(start[1], end[1] + 1):
                self.decks.append(Deck(coord_row, column))
```
Try to do code simple as possible